src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp

```diff
@@ -217,15 +233,16 @@ bool PCRPrimerDesignForDNAAssemblyTask::areMetlingTempAndDeltaGood(const QByteAr
 }
 
 bool PCRPrimerDesignForDNAAssemblyTask::hasUnwantedConnections(const QByteArray& primer) const {
-    bool isUnwantedSelfDimer = UnwantedConnectionsUtils::isUnwantedSelfDimer(primer, settings.gibbsFreeEnergyExclude, settings.meltingPointExclude, settings.complementLengthExclude);
+    QString unused;
```

Comment:
I suggest you move this parameter to `UnwantedConnectionsUtils`, not messing it here. It will be two functions like this:

```
static bool isUnwantedHeteroDimer(const QByteArray& forwardSequence,
                                  const QByteArray& reverseSequence,
                                  double unwantedDeltaG,
                                  double unwantedMeltingTemperature,
                                  int unwantedDimerLength,
                                  QString &report) {
    //... do smth
}

static bool isUnwantedHeteroDimer(const QByteArray& forwardSequence,
                                  const QByteArray& reverseSequence,
                                  double unwantedDeltaG,
                                  double unwantedMeltingTemperature,
                                  int unwantedDimerLength) {

    QString unused;
    return isUnwantedHeteroDimer(...., unused);
}
```